saleor/graphql/meta/types.py

```diff
@@ -0,0 +1,54 @@
+import graphene
+
+from ...core.models import ModelWithMetadata
+from .resolvers import (
+    resolve_metadata,
+    resolve_object_with_metadata_type,
+    resolve_private_metadata,
+)
+
+
+class MetaItem(graphene.ObjectType):
+    key = graphene.String(required=True, description="Key for stored data.")
+    value = graphene.String(required=True, description="Stored metadata value.")
+
+
+class ObjectWithMetadata(graphene.Interface):
+    private_metadata = graphene.List(
+        MetaItem,
+        required=True,
+        description="List of publicly stored metadata namespaces.",
```

Comment:
This one is not public and we don't have explicit namespaces anymore. 
```suggestion
        description="List of private metadata items. Requires proper staff permissions to access.",
```